lib/matplotlib/figure.py

```diff
@@ -2044,9 +2044,12 @@ def savefig(self, fname, *, transparent=None, **kwargs):
             possibly some backend-dependent object such as
             `matplotlib.backends.backend_pdf.PdfPages`.
 
-            If *format* is not set, then the output format is inferred from
-            the extension of *fname*, if any, and from :rc:`savefig.format`
-            otherwise.  If *format* is set, it determines the output format.
+            If *format* is not set, then the output format is inferred 1) from
+            the extension of *fname*, if any, and 2) from :rc:`savefig.format`
+            otherwise.  In this last case (2) only, the extension is appended
+            to *fname*.
+
+            If *format* is set, it determines the output format.
```

Comment:
Suggest:  

```python
If *format* is set, it determines the output format, and the file is saved as *fname*.  Note that *fname* is used verbatim, and there is no attempt to make the extension, if any, of *fname* match *format*, and no extension is appended.  

If *format* is not set, then the format is inferred from the extension of *fname*, if there is one.  If *format* is not set and *fname* has no extension, then the file is saved with `savefig.format` and the appropriate extension is appended to *fname*. 
```

Question: What happens if I have an extension that is not recognized?  i.e. 'myfigure.0034`? Does it fall back?

Reply:
You get an error "don't know how to save a .0034 file".
Again, I don't think this behavior is particularly good, but I'm not trying to change this here.

I like your wording, put it in.